src/database/queries.js

```diff
@@ -0,0 +1,912 @@
+            to_time:   s.to_time,
+            reason:    s.reason,
+            user_id:   s.user_id
+        }));
+    });
+
+    // return getUser(discord_user_id).then(user => {
+    // const filter = { where: { user_id: user.id } };
+    //     return UnavailableSchedule.findAll(filter);
+    // });
+};
+
+/**
+ * Get a list of all unavailable dates
+ * @returns 
+ */
+const getAllUnavailable = async () => {
+
+    // TODO: figure out a way to filter out past dates
+    // TODO: figure out a way to delete past schedules
+
+    return UnavailableSchedule.findAll();
+};
+
+/**
+ * Soft deletes all expired unavailable schedules
+ */
+const deleteExpiredUnavailable = async () => {
+    return UnavailableSchedule.destroy({ where: { to_time: { [Op.lte]: dayjs().utc().format() } } });
+};
+
+/**
+ * Add an unavailable schedule to a user
+ * @param {{
+* id: string
+* from: string,
+* to: string,
+* days: string[]}} schedule required
+*/
+const setAvailable = async (schedule) => {
+    const discord_user_id = schedule?.id?.toString()?.trim() ?? '';
+    const from_time = schedule?.from?.toString()?.trim() ?? '';
+    const to_time = schedule?.to?.toString()?.trim() ?? '';
+    const days = schedule?.days?.map(day => day.toString().trim().toLocaleLowerCase());
+
+    assertArgument(discord_user_id.length > 0, 'Invalid Argument: schedule.id');
+    assertArgument(from_time.length > 0, 'Invalid Argument: schedule.from');
+    assertArgument(to_time.length > 0, 'Invalid Argument: schedule.to');
+    assertArgument(days.length > 0, 'Invalid Argument: schedule.days');
+
+    // create or update
+
+    // make sure to set the user_id foreign key
+    return getUser(discord_user_id)
+        .then(user => {
+            const filter = { where: { user_id: user.id }, };
+            return AvailableSchedule.findOne(filter)
+                .then(available => {
+                    if (available) return available.update({
+                        from_time,
+                        to_time,
+                        days
+                    });
+                    return AvailableSchedule.create({
+                        user_id: user.id,
+                        from_time,
+                        to_time,
+                        days
+                    });
+                });
+        });
+};
+
+/**
+* Get a list of unavailable dates for a user 
+* @param {string} userId required
+* @returns 
+*/
+const getAvailable = async (userId) => {
+    const discord_user_id = userId?.toString()?.trim() ?? '';
+
+    assertArgument(discord_user_id.length > 0, 'Invalid Argument: userId');
+
+    // TODO: figure out a way to filter out past dates
+    // TODO: figure out a way to delete past schedules
+
+    return getModelEntryByUser(discord_user_id, AvailableSchedule);
+
+    // return getUser(discord_user_id).then(user => {
+    //     const filter = { where: { user_id: user.id } };
+    //     return AvailableSchedule.findOne(filter);
+    // }) 
```

Comment:
Some junk like these comments in this file. If it's a time sink then skip cleaning it up, but I'll want to follow up with a cleanup PR.